### PR TITLE
Add config to build Mixer release on GCB

### PR DESCRIPTION
**Pull request by rvkubiak**
**Release note**:
```release-note
Add a config file for running Mixer release builds on Google Container Builder.
```

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/1123)

### CODE REVIEW

File: bin/build_mixer_release.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Used for automated builds triggered on Git tags pushed matching the regex:
+#   \d+\.\d+\.\d+
+
+steps:
+- name: 'gcr.io/$PROJECT_ID/istio-builder'
+  args: [ 'bash', './bin/publish-docker-images.sh', '-h', "gcr.io/$PROJECT_ID", '-t', "$TAG_NAME" ]
```
Comment (sebastienvas) [inline bin/build_mixer_release.yaml]: use gcr.io/istio-io,docker.io/istio for hubs

Comment (rvkubiak) [inline bin/build_mixer_release.yaml]: Hard coded the project names.  Will add docker.io once credential work is finished.

Comment (rvkubiak) [inline bin/build_mixer_release.yaml]: Added release script and an DockerHub push.

**Comment by sebastienvas** [comment]
Need approval from someone else. Added @ZackButcher

**Pull request merged**